content/en/real_user_monitoring/guide/monitor-hybrid-react-native-applications.md

```diff
@@ -0,0 +1,323 @@
+```
+
+{{< tabs >}}
+{{% tab "Android" %}}
+
+Add the Datadog Android SDK to your dependencies in your `android/app/build.gradle` file:
+
+```gradle
+// The version is set by @datadog/mobile-react-native
+implementation "com.datadoghq:dd-sdk-android-rum"
+implementation "com.datadoghq:dd-sdk-android-logs"
+implementation "com.datadoghq:dd-sdk-android-trace"
+implementation "com.datadoghq:dd-sdk-android-webview"
+```
+
+Initialize the SDK on the native side. See the official documentation for [Android][1] for instructions.
+
+[1]: /real_user_monitoring/mobile_and_tv_monitoring/setup/android/?tab=kotlin
+
+{{% /tab %}}
+{{% tab "iOS" %}}
+
+Initialize the SDK on the native side. See the official documentation for [iOS][1] for instructions.
+
+[1]: /real_user_monitoring/mobile_and_tv_monitoring/setup/ios/?tab=cocoapods
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### Instrumenting React Native RUM Views
+
+{{< tabs >}}
+{{% tab "Android" %}}
+
+Use a `ComponentPredicate` to filter out native views created by your navigation libraries:
+
+```kotlin
+// Adapt the Fragment type to your View tracking strategy
+class RNComponentPredicate : ComponentPredicate<Fragment> {
+    override fun accept(component: Fragment): Boolean {
+        // Identify and drop react native screen views
+        if (component.javaClass.name.startsWith("com.swmansion.rnscreens")) {
+            return false
+        }
+        if (component.javaClass.name.startsWith("com.facebook.react")) {
+            return false
+        }
+        return true
+    }
+
+    override fun getViewName(component: Fragment): String? {
+        return null
+    }
+}
+
+// Use it in your RUM configuration
+rumConfiguration.useViewTrackingStrategy(FragmentViewTrackingStrategy(true, RNComponentPredicate()))
+```
+Then, use `@datadog/mobile-react-navigation` to track your views.
+
+If you have enabled ProGuard obfuscation, add rules to prevent obfuscation of the target packages in release builds.
+
+{{% /tab %}}
+{{% tab "iOS" %}}
+
+Use a `UIKitRUMViewsPredicate` to filter out native views created by your navigation libraries:
+
+```swift
+class RNHybridPredicate: UIKitRUMViewsPredicate {
+    var defaultPredicate = DefaultUIKitRUMViewsPredicate()
+
+    func rumView(for viewController: UIViewController) -> RUMView? {
+        let canonicalClassName = NSStringFromClass(type(of: viewController))
+        // Dropping RN Views
+        if (canonicalClassName.starts(with: "RN")) {
+            return nil
+        }
+
+        return defaultPredicate.rumView(for: viewController)
+    }
+}
+
+// Use it in your RUM configuration
+let rumConfiguration = RUM.Configuration(
+    applicationID: applicationId,
+    uiKitViewsPredicate: RNHybridPredicate(),
+)
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+Then, use `@datadog/mobile-react-navigation` to track your views.
+
+If you have enabled ProGuard obfuscation, add rules to prevent obfuscation of the target packages in release builds.
```

Comment:
I think this was meant to be inside the iOS tab

Reply:
I don't think this is resolved. It still shows twice when the Android tab is selected. It either shouldn't be in that tab, or this instance should go in the iOS tab.

Reply:
Ahh, it's a duplicate. Will delete.